src/cmd/create.go

```diff
@@ -710,19 +710,50 @@ func pullImage(image, release, authFile string) (bool, error) {
 		shouldPullImage = true
 	}
 
+	imageSizeChan := make(chan string)
+	confirmationChan := make(chan bool)
+	var userResponded bool
+
 	if promptForDownload {
 		fmt.Println("Image required to create toolbox container.")
 
-		var prompt string
+		prompt := fmt.Sprintf("Download %s (...)? [y/N]:", imageFull)
+		go func() {
+			shouldPullImage = askForConfirmation(prompt)
+			confirmationChan <- shouldPullImage
+		}()
 
-		if imageSize, err := getImageSizeFromRegistry(imageFull); err != nil {
-			logrus.Debugf("Getting image size failed: %s", err)
-			prompt = fmt.Sprintf("Download %s? [y/N]:", imageFull)
-		} else {
-			prompt = fmt.Sprintf("Download %s (%s)? [y/N]:", imageFull, imageSize)
+		// Launch a goroutine to call getImageSizeFromRegistry and send the size to the channel
+		go func() {
+			imageSize, err := getImageSizeFromRegistry(imageFull)
+			if err != nil {
+				logrus.Debugf("Getting image size failed: %s", err)
+				imageSize = ""
+			}
+			imageSizeChan <- imageSize
+		}()
+
+		var imageSize string
+		select {
+		case imageSize = <-imageSizeChan:
+			if imageSize != "" {
+				prompt = fmt.Sprintf("\rDownload %s (%s)? [y/N]:", imageFull, imageSize)
```

Comment:
The race is here.  We have 3 threads of execution - the two goroutines above and the *main* thread.  By the time we reach here, we can only be sure that the second goroutine attempted to get the size from the registry with `skopeo inspect`, but there's no guarantee that the first goroutine has shown the prompt for the first time already.

This is unlikely to happen, because in most cases running `skopeo inspect` will be so slow that the first goroutine to show the prompt will get scheduled to run on the CPU before `skopeo inspect` returns.  However, it can still happen if the stars align against us.  :)

To avoid this uncertainty, we need to cancel the first `askForConfirmation` call, and then start a second one.  This is why I wrote the *cancellable* asynchronous read [example](https://rishi.fedorapeople.org/async-read.go) for you.  Here, we need to create an `askForConfirmationAsync` analogous to the `scanFileAsync` in the example.